cmake/extensions.cmake

```diff
@@ -375,6 +375,49 @@ macro(zephyr_library_named name)
   target_link_libraries(${name} PUBLIC zephyr_interface)
 endmacro()
 
+# Determines what the current directory's lib name would be and writes
+# it to the argument "lib_name" as if ZEPHYR_BASE had been equal to
+# ZEPHYR_CURRENT_MODULE_DIR.
+macro(zephyr_library_amend_module_lib_name lib_name)
+  if(NOT ZEPHYR_CURRENT_MODULE_DIR)
+    message(FATAL_ERROR "Function only available for Zephyr modules.")
+  endif()
+  # Remove the prefix (/path/to/<ZEPHYR_MODULE>/driver/serial/CMakeLists.txt => driver/serial/CMakeLists.txt)
+  file(RELATIVE_PATH name ${ZEPHYR_CURRENT_MODULE_DIR} ${CMAKE_CURRENT_LIST_FILE})
+
+  # Remove the filename (driver/serial/CMakeLists.txt => driver/serial)
+  get_filename_component(name ${name} DIRECTORY)
+
+  # Replace / with __ (driver/serial => driver__serial)
+  string(REGEX REPLACE "/" "__" name ${name})
+
+  set(${lib_name} ${name})
+endmacro()
```

Comment:
Let's increase the code re-use with zephyr_library_get_current_dir_lib_name

Reply:
agree, and done.